topics/dev/tutorials/tool-from-scratch/tutorial.md

```diff
@@ -0,0 +1,1293 @@
+groups of parameters with similar functions. Alternatively, it can hide "advanced" or optional parameters, so that the user doesn't have to deal with
+otherwise-irrelevant parameters if they don't need to. There are only 4 attributes:
+
+- Name
+  - Sets the internal reference name for the section. Cannot contain spaces.
+- Title
+  - Sets the visible name of the section for the user. Can contain spaces.
+- Expanded
+  - A boolean value that sets the default starting position of the section. If false, the section will begin collapsed, and only the title will be visible by default.
+- Help
+  - Help text visible to the user that explains the purpose of the section.
+
+#### *Inputs Section in Practice*
+
+The bellerophon tool's help section reads
+
+```txt
+optional arguments:
+  -h, --help            show this help message and exit
+  --forward FORWARD, -f FORWARD
+                        SAM/BAM/CRAM file with the first set of reads.
+  --reverse REVERSE, -r REVERSE
+                        SAM/BAM/CRAM file with the second set of reads.
+  --output OUTPUT, -o OUTPUT
+                        Output BAM file for filtered and paired reads.
+  --quality QUALITY, -q QUALITY
+                        Minimum mapping quality.
+  --threads THREADS, -t THREADS
+                        Threads.
+  --log-level {CRITICAL,ERROR,WARNING,INFO,DEBUG}, -l {CRITICAL,ERROR,WARNING,INFO,DEBUG}
+                        Log level.
+  --version             show program's version number and exit
+```
+
+This means that there are 6 total functional flags, --forward, --reverse, --output, --quality, --threads and --log-level.
+Among these flags, two are file inputs, or 'data' type params, one is an integer type param, and three are parameters not
+necessary for the user to have direct access to. As Galaxy uses specific file names or patterns to find output files, the
+output file names need to be specific to the wrapper, not the actual running step. Since the user doesn't know the capability
+of the resources on which the tool will be running, they should not be given access to the threads flag. And since the log level
+goes to the stderr instead of an output file, this can also be skipped. All three of these flags can be coded in the command,
+but not exposed to the user.
+
+> ### {% icon hands_on %} Hands-on: Adding parameters to your tool
+> 1. Using information in the above sections, add parameters for the forward, reverse, and quality flags to your tool xml.
+>
+> > ### {% icon question %} Question
+> >
+> > What does your `<inputs/>` section look like?
+> >
+> > > ### {% icon solution %} Solution
+> > > To include all of the necessary parameters, then, the inputs section would appear like this
+> > >
+> > > ```xml
+> > > <inputs>
+> > >     <param argument="--forward" type="data" format="sam" label="First set of reads" help="This is usually the forward reads in your experiment." />
```

Comment:
@astrovsky01 should this be?

```suggestion
> > >     <param argument="--forward" type="data" format="sam,bam" label="First set of reads" help="This is usually the forward reads in your experiment." />
```

Reply:
Fixed with "qname_sorted.bam,sam". Good catch, thanks!